autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -201,6 +202,46 @@ fn to_string(value: Vec<u8>, name: &str) -> Result<String, DbError> {
     })
 }
 
+/// Parse the "set" (see [DbClient::add_channels]) of channel ids in a bigtable Row.
+///
+/// Cells should solely contain the set of channels otherwise an Error is returned.
+fn channels_from_cells(cells: &RowCells) -> DbResult<HashSet<Uuid>> {
+    let mut result = HashSet::new();
+    for cells in cells.values() {
+        let Some(cell) = cells.last() else {
+            continue;
+        };
+        let Some((_, chid)) = cell.qualifier.split_once("chid:") else {
+            return Err(DbError::Integrity(
+                "get_channels expected: chid:<chid>".to_owned(),
+                None,
+            ));
+        };
+        result.insert(Uuid::from_str(chid).map_err(|e| DbError::General(e.to_string()))?);
+    }
+    Ok(result)
+}
+
+/// Convert the [HashSet] of channel ids to cell entries for a bigtable Row
+fn channels_to_cells(channels: Cow<HashSet<Uuid>>, expiry: SystemTime) -> Vec<cell::Cell> {
+    let channels = channels.into_owned();
+    let mut cells = Vec::with_capacity(channels.len().min(100_000));
+    for (i, channel_id) in channels.into_iter().enumerate() {
```

Comment:
Is there a desire to check the size of `challens` prior to iteration, or is it intended to be a processing limit at runtime while looping through? Just curious in case a check before iterating was valuable.

Reply:
Yes, we're going to begin enforcing a max limit of channels shortly, which will probably end up changing this check https://github.com/mozilla-services/autopush-rs/issues/733